### PR TITLE
Spring Exercise 1 - Homework 6

### PR DESCRIPTION
**Pull request by asish918**
# Homework 6 Asish

### Topics Covered:
- Spring Context
- Spring Beans
- Spring Annotations

### Output Screenshots - 
The Vehicle objects are different. Have a look at their prices. I used a for loop and hardcoded the Tyre and Speaker Brands that's the reason everything appears to be similar.
Screenshot 2024-01-22 at 21 52 06

### CODE REVIEW

**Comment by asish918** [comment]
| Metric | Value |
| ------ | ----- |
| alert_status | <b>:white_check_mark: OK</b> |
| bugs | 0 |
| code_smells | 1 |
| reliability_rating | A (0 Bugs)<br/>Note: A being best, E being worst |
| security_rating | A (0 Vulnerabilities)<br/>Note: A being best, E being worst |
| vulnerabilities | 0 |

File: backend/homework/6/src/main/java/asish/kdu/Main.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+     * INFO, DEBUG tagged logs being printed to the console. Refer the logback.xml file in resources folder
+     */
+    public static void main(String[] args) {
+        AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext(AppConfig.class);
+        VehicleService vhs = context.getBean(VehicleService.class);
+        try {
+            List<Vehicle> listV = vhs.generateVehicleList(5);
+            CustomLogger.printLogger("-------------\nVehicle List - \n", CustomLogger.LoggerType.INFO);
+            listV.forEach(v -> CustomLogger.printLogger(v.toString(), CustomLogger.LoggerType.INFO));
+        } catch (SpeakerException | TyreException | VehicleException e) {
```
Comment (rishav-kd) [inline backend/homework/6/src/main/java/asish/kdu/Main.java:27]: Lets have different catch blocks for different exceptions.

File: backend/homework/6/src/main/java/asish/kdu/services/TyreService.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+     * Generates and returns appropriate Tyre object.
+     * @return {@link  asish.kdu.entities.Tyre Tyre} object
+     * @param brand Enum of TyreBrand
+     * @throws TyreException When invalid value is passed
+     */
+    public Tyre generateTyre(TyreBrand brand) throws TyreException {
+        if (brand.equals(TyreBrand.BRIDGESTONE)) {
+            return new Tyre(TyreBrand.BRIDGESTONE, 150);
+        } else if (brand.equals(TyreBrand.MRF)) {
+            return new Tyre(TyreBrand.MRF, 120);
```
Comment (rishav-kd) [inline backend/homework/6/src/main/java/asish/kdu/services/TyreService.java:24]: a better way would have been to extend the enum object to include price as well.

File: backend/homework/6/src/main/java/asish/kdu/logging/ColouredLayout.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+ * It decides color based on Different levels of logging
+ */
+public class ColouredLayout extends LayoutBase<ILoggingEvent> {
+
+    public String doLayout(ILoggingEvent event) {
+        StringBuffer sbuf = new StringBuffer(128);
+
+        switch (event.getLevel().toInt()) {
+            case Level.ERROR_INT:
+                sbuf.append("\u001B[31m"); // Red for ERROR
```
Comment (rishav-kd) [inline backend/homework/6/src/main/java/asish/kdu/logging/ColouredLayout.java:21]: lets add these as constants

**Review by rishav-kd — COMMENTED** [review]
Good. but missing the @Bean annotation. Otherwise everything looks good.

**Pull request closed without merging**